examples/forms-signup.rs

```diff
@@ -261,26 +297,54 @@ fn logged_in(username: &str, app_state: &Dynamic<AppState>) -> impl MakeWidget {
 }
 
 fn handle_login(
-    username: String,
-    password: MaskedString,
+    login_args: LoginArgs,
     api: &channel::Sender<FakeApiRequest>,
     app_state: &Dynamic<AppState>,
     form_state: &Dynamic<NewUserState>,
-    api_errors: &Dynamic<Map<SignupField, String>>,
+    form_errors: &Dynamic<Map<SignupFormField, String>>,
 ) {
-    let response = FakeApiRequestKind::SignUp {
-        username: username.clone(),
-        password,
-    }
-    .send_to(api);
+    let request = FakeApiRequestKind::SignUp {
+        username: login_args.username.clone(),
+        password: login_args.password,
+    };
+
+    let response = request
+        .send_to(api);
+
     match response {
         FakeApiResponse::SignUpSuccess => {
-            app_state.set(AppState::LoggedIn { username });
+            app_state.set(AppState::LoggedIn { username: login_args.username });
             form_state.set(NewUserState::Done);
         }
-        FakeApiResponse::SignUpFailure(errors) => {
+        FakeApiResponse::SignUpFailure(mut errors) => {
             form_state.set(NewUserState::FormEntry);
-            api_errors.set(errors);
+
+            // match up the API errors to form errors, there may not be a 1:1 relationship with form fields and api errors
+            let mut mapped_errors: Map<SignupFormField, String> = Default::default();
+
+            for code in errors.drain(..).into_iter() {
+                match code.try_into() {
+                    Ok(FakeApiSignupErrorCode::UsernameReserved) |
+                    Ok(FakeApiSignupErrorCode::UsernameUnavailable)
+                    => {
+                        // handle the two cases with the same error message
+                        mapped_errors.insert(SignupFormField::Username, String::from("Username is a unavailable"));
+                    },
+                    Ok(FakeApiSignupErrorCode::UsernameInvalid)
+                    => {
+                        mapped_errors.insert(SignupFormField::Username, String::from("Username is invalid"));
```

Comment:
If we're going to include this as an example, error messages should contain the information required to pass validation.